### ISSUE TITLE
modules/beta-autopilot-private-cluster being replaced with every terraform apply

### ISSUE THREAD

**Issue by dgnemo** [terraform-google-modules/terraform-google-kubernetes-engine#1713]
### TL;DR

I am trying to use `terraform-google-modules/kubernetes-engine/google/modules/beta-autopilot-private-cluster` 

But every `terraform apply` invocation returns:
```
module.gke.google_container_cluster.primary must be replaced
 ```

### Expected behavior

After creating the cluster, terraform should not destroy the cluster at each subsequent `terraform apply`

### Terraform Configuration

```hcl
module "gke" {
  source                        = "terraform-google-modules/kubernetes-engine/google/modules/beta-autopilot-private-cluster"
  version                       = "27.0.0"
  project_id                    = google_project.corp_k8s.project_id
  name                          = "corp-k8s"
  region                        = "us-central1"
  zones                         = ["us-central1-a", "us-central1-b", "us-central1-f"]
  network_project_id            = data.google_project.corp_vpc.project_id
  network                       = data.google_compute_network.corp_vpc.name
  subnetwork                    = data.google_compute_subnetwork.corp_vpc_us.name
  master_ipv4_cidr_block        = "10.4.0.0/28"
  ip_range_pods                 = "corp-k8s-pods"
  ip_range_services             = "corp-k8s-services"
  enable_private_endpoint       = true
  enable_private_nodes          = true
  deploy_using_private_endpoint = true
  enable_cost_allocation        = true
  master_authorized_networks = [
    {
      cidr_block   = "10.2.0.0/20"
      display_name = "corp-vpc-us"
    },
    {
      cidr_block   = "10.3.0.0/20"
      display_name = "corp-vpc-uk"
    },
  ]
  depends_on = [
    google_project_service.corp_k8s_google_services["container.googleapis.com"],
  ]
}
```
```


### Terraform Version

```sh
Terraform v1.5.5
on linux_amd64
+ provider registry.terraform.io/gavinbunney/kubectl v1.14.0
+ provider registry.terraform.io/hashicorp/google v4.78.0
+ provider registry.terraform.io/hashicorp/google-beta v4.78.0
+ provider registry.terraform.io/hashicorp/helm v2.10.1
+ provider registry.terraform.io/hashicorp/kubernetes v2.23.0
+ provider registry.terraform.io/hashicorp/random v3.5.1
```

**Comment by pndurette** [comment]
Not sure if that's the same issue, but I'm hitting this as well, the part that forces replacement is:

```hcl
- dns_config { # forces replacement
     - cluster_dns        = "CLOUD_DNS" -> null
     - cluster_dns_domain = "cluster.local" -> null
     - cluster_dns_scope  = "CLUSTER_SCOPE" -> null
   }
```

However in the `beta-autopilot-private-cluster`, the `dns_config` config doesn't get surfaced to the module variables in any way.

This likely has something to do with the message the Console displays:
> Starting from September 2023, all GKE Autopilot clusters (minimum version to be determined in August 2023) will migrate to Cloud DNS as the default DNS provider (in replacement of kube-dns), at no extra charge.

**Comment by dgnemo** [comment]
I did not see it before because of the amount of output generated by terraform, but yes, the part that forces the replacement is the same for me too:

```
      - dns_config { # forces replacement
          - cluster_dns        = "CLOUD_DNS" -> null
          - cluster_dns_domain = "cluster.local" -> null
          - cluster_dns_scope  = "CLUSTER_SCOPE" -> null
        }
```

**Comment by pndurette** [comment]
To add—
I did some testing: even when agree to have the cluster rebuilt, it will still want to rebuilt it on the next `apply`.

So we need a change at the `dns_config` level to match the new DNS defaults otherwise this breaks that module entirely.

**Comment by AuroraLi** [comment]
I am also seeing this issue

**Comment by chy168** [comment]
also encountered the same issue, same force replacement in output:
```
# module.gke.google_container_cluster.primary must be replaced
-/+ resource "google_container_cluster" "primary" {

...

}

.
.
.

- dns_config { # forces replacement
    - cluster_dns        = "CLOUD_DNS" -> null
    - cluster_dns_domain = "cluster.local" -> null
    - cluster_dns_scope  = "CLUSTER_SCOPE" -> null
  }
```

**Comment by micahtmiller** [comment]
It looks like a fix is in process, hopefully we will see that merged soon.

Question to the experts, would a "lifecycle {ignore_changes}" on the dns_config attribute also work? 

Is there any reason to not allow the consumer of this module to send in their own lifecycle definition?

**Comment by dgnemo** [comment]
@micahtmiller would not call myself an expert of terraform, but I think `lifecycle {ignore_changes}` cannot help us here because the `module "name" {...}`  does not allow `lifecyle` attribute, only `resource "provider" "name" {...}` can use that feature.

**Comment by bharathkkb** [comment]
@dgnemo is this still happening with the latest provider release `4.80.0`? It seems to include a potential fix https://github.com/hashicorp/terraform-provider-google/pull/15549
